docs/content/users/install/ddev-installation.md

```diff
@@ -266,6 +266,48 @@ Installing and upgrading DDEV are nearly the same thing, because you're upgradin
 
     It can be complicated to get private databases and files into Gitpod, so in addition to the launchers, the [`git` provider example](https://github.com/drud/ddev/blob/master/pkg/ddevapp/dotddev_assets/providers/git.yaml.example) demonstrates pulling a database and files without complex setup or permissions. This was created explicitly for Gitpod integration, because in Gitpod you typically already have access to private Git repositories, which are a fine place to put a starter database and files. Although [ddev-gitpod-launcher](https://drud.github.io/ddev-gitpod-launcher/) and the web extension provide the capability, you may want to integrate a Git provider—or one of the [other providers](https://github.com/drud/ddev/tree/master/pkg/ddevapp/dotddev_assets/providers)—for each project.
 
+=== "Codespaces"
+
+    ## GitHub Codespaces
+
+    DDEV is fully supported in [GitHub Codespaces](https://github.com/features/codespaces)
+
+    To run DDEV in codespaces you'll need to create a codespaces instance for your repo and add a `.devcontainer/devcontainer.json` to it as shown below. There are at least two ways to do this:
+
+    1. On your project repo on github.com click `Code → Codespaces → ... → Configure dev container` to create or edit a `devcontainer.json`. Edit or create the file to add the features shown in the example.
+        ![codespaces dev container](../../../images/codespaces_devcontainer.jpg)
+
+    2. Open your project in Codespaces and edit the `.devcontainer/devcontainer.json`, then rebuild the container, using VS Code’s “Rebuild container” action. (<kbd>⌘</kbd> + <kbd>SHIFT</kbd> + <kbd>P</kbd> on a Mac or <kbd>CTRL</kbd> + <kbd>SHIFT</kbd> + <kbd>P</kbd> on Windows, then search for “rebuild”.)
```

Comment:
Reword, turn number into bullet:

```suggestion
    * Open your project’s codespace directly, edit the `.devcontainer/devcontainer.json` file, and rebuild the container with VS Code’s “Rebuild container” action. (<kbd>⌘</kbd> + <kbd>SHIFT</kbd> + <kbd>P</kbd> on a Mac or <kbd>CTRL</kbd> + <kbd>SHIFT</kbd> + <kbd>P</kbd> on Windows, then search for “rebuild”.)
```

This still isn’t working for me; no `ddev` in the codespace’s Terminal and my made-up “Rebuild container” action doesn’t exist. Closest guesses are “Docker Containers: Restart” and “Docker Containers: Refresh” which don’t seem to do anything.

Reply:
Would be happy to take a look at this with you if you have a moment. Thanks for testing!